delira/data_loading/dataset.py

```diff
@@ -233,167 +225,134 @@ def __len__(self):
-        img_extensions : list
-            valid extensions of image files
-        gt_extensions : list
-            valid extensions of label files
+            function to load a single data sample
         **load_kwargs :
             additional loading keyword arguments (image shape,
-            channel number, ...); passed to load_fn
+            channel number, ...); passed to _sample_fn
 
         """
-        super().__init__(data_path, load_fn, img_extensions, gt_extensions)
+        super().__init__(data_path, load_fn)
         self._load_kwargs = load_kwargs
-        self.data = self._make_dataset(self.data_path)
+        self.data = self._make_dataset(data_path)
 
-    def _make_dataset(self, path):
+    def _make_dataset(self, path: typing.Union[str, list]):
         """
-        Helper Function to make a dataset containing paths to all images in a
-        certain directory
+        Helper Function to make a dataset containing all samples in a certain
+        directory
 
         Parameters
         ----------
-        path : str
-            path to data samples
+        path: str or list
+            if data_path is a string, _sample_fn is called for all items inside
+            the specified directory
+            if data_path is a list, _sample_fn is called for elements in the
+            list
 
         Returns
         -------
         list
-            list of sample paths
+            list of items which where returned from _sample_fn (typically dict)
 
         Raises
         ------
         AssertionError
-            if `path` is not a valid directory
+            if `path` is not a list and is not a valid directory
 
         """
         data = []
-        assert os.path.isdir(path), '%s is not a valid directory' % dir
-
-        for root, _, fnames in sorted(os.walk(path)):
-            for fname in fnames:
-                fpath = os.path.join(root, fname)
-                if self._is_valid_image_file(fpath):
-                    sample = [fpath]
-                    for ext in self._gt_extensions:
-                        gt_path = fpath.rsplit(".", maxsplit=1)[0] + ext
-                        if os.path.isfile(gt_path):
-                            sample.append(gt_path)
-
-                    data.append(sample)
-
+        if isinstance(path, list):
+            # iterate over all elements
+            for p in tqdm(path, unit='samples', desc="Loading samples"):
+                data.append(self._load_fn(p, **self._load_kwargs))
+        else:
+            # call _sample_fn for all elements inside directory
+            assert os.path.isdir(path), '%s is not a valid directory' % dir
+            for p in tqdm(os.listdir(path), unit='samples',
+                          desc="Loading samples"):
+                data.append(self._load_fn(os.path.join(path, p),
+                                          **self._load_kwargs))
         return data
 
-    def _is_valid_image_file(self, fname):
```

Comment:
Maybe we want to include this (or something similar) as a helper function in the utilities?

Reply:
Do you have a concrete idea where/how I could add this? The new datasets do not use `_img_extensions` as they were previously implemented.

Reply:
I'd just leave them as a variable so they have to be passed explicitly.